src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1807,7 +1829,11 @@ FixedwingPositionControl::control_auto_landing(const hrt_abstime &now, const flo
 
 			} else {
 				// normal navigation
-				_l1_control.navigate_waypoints(prev_wp, curr_wp, curr_pos, ground_speed);
+				Vector2f curr_pos_local{_local_pos.x, _local_pos.y};
+				Vector2f curr_wp_local = _global_local_proj_ref.project(curr_wp(0), curr_wp(1));
+				Vector2f prev_wp_local = _global_local_proj_ref.project(prev_wp(0),
```

Comment:
again, can these lines be done once at top of method?

Reply:
This would mean that even if we are not using the local positions, we will be projecting them into local coordinates. Does this make it better than the current state?

Reply:
i guess the better way in general here would be refactoring this landing section not to have so much loaded into one method. i was only thinking in terms of this intermediate step (assuming refactoring is coming later) that for now we don't copy paste things that are used a few times too much, as it's prone to bugs later once they get moved.

for the current PR i dont have very strong feelings about this.. just thought it would be a little more concise, even if at the expense of an extra projection. but, as devil's advocate.. 1. is it that expensive of an operation? and 2. we anyway need to be able to afford that calculation every iteration if we are in auto mode, so not sure what we are saving. -- also we would get rid of the extra double defined at top if immediately converted
https://github.com/PX4/PX4-Autopilot/blob/9b3b013cca964f84810141d43cb004b3b347a681/src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp#L1633-L1648

your call.

Reply:
I was not referring to performance gains, but more on the side of being explicit on where positions are used versus not. Since some fw landing states do not use position at all, I thought this will help in future refactors. However, since I also don't have strong preferences, have moved the projections into the top of the method.

Reply:
ok - we will put them in better places with the coming refactor :) thanks for taking the time to start thinking about these things!